File: Sample Code/ObjcSampleCode/DJISdkDemo/Demo/DJIMapView/DJIAnnotationViews/DJIPOIAnnotationView.m
//
//  DJIPOIAnnotationView.m
//  Phantom3
//
//  Created by Jerome.zhang on 15/7/30.
//  Copyright (c) 2015年 DJIDevelopers.com. All rights reserved.
//

#import "DJIPOIAnnotationView.h"

@interface DJIPOIAnnotationView ()

@end

@implementation DJIPOIAnnotationView
- (id)initWithAnnotation:(id<MKAnnotation>)annotation reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithAnnotation:annotation reuseIdentifier:reuseIdentifier];
    if (self) {
        self.pinView = [[UIView alloc] initWithFrame:CGRectMake(0.0, 0.0, 54.0, 54.0)];
        [self.pinView setBackgroundColor:[UIColor clearColor]];
        UIImageView *pinIcon = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"navigation_poi_pin"]];
        [self.pinView addSubview:pinIcon];
        [pinIcon setCenter:CGPointMake(self.pinView.center.x, self.pinView.center.y - pinIcon.frame.size.height*0.5)];
        CGRect frame = self.frame;
        frame.size = self.pinView.frame.size;
        [self setFrame:frame];
        [self addSubview:self.pinView];
        self.pinView.layer.zPosition = 101;
    }
    
    return self;
}


@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Missions/DJIMapView/DJIAnnotationViews/DJIPOIAnnotationView.swift
//
//  DJIPOIAnnotationView.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 15/7/30.
//  Copyright (c) 2015 DJIDevelopers.com. All rights reserved.
//
import MapKit

class DJIPOIAnnotationView: MKAnnotationView {
    var pinView: UIView?=nil

   override init (annotation: MKAnnotation?, reuseIdentifier: String?) {
    
        super.init(annotation: annotation, reuseIdentifier: reuseIdentifier)
        if (pinView == nil) {
            pinView = UIView(frame: CGRect(x: 0.0, y: 0.0, width: 54.0, height: 54.0))
            pinView!.backgroundColor = UIColor.clear
            let pinIcon: UIImageView = UIImageView(image: UIImage(named: "navigation_poi_pin"))
            pinView!.addSubview(pinIcon)
            pinIcon.center = CGPoint(x: self.pinView!.center.x, y: self.pinView!.center.y - pinIcon.frame.size.height * 0.5)
        }

        var frame: CGRect = self.frame
        frame.size = pinView!.frame.size
        self.frame = frame
        self.addSubview(pinView!)
        pinView!.layer.zPosition = 101
    }

   required init?(coder aDecoder: NSCoder) {
       super.init(coder: aDecoder)
   }
}
